memphis/consumer.py

```diff
@@ -156,6 +159,18 @@ async def main(/, host, username, password, station):
         
         """
         messages = []
+
+        if prefetch and self.cached_messages:
+            if len(self.cached_messages) >= batch_size:
+                messages = self.cached_messages[:batch_size]
+                self.cached_messages = self.cached_messages[batch_size:]
+            else:
+                pulled_messages_size = len(self.cached_messages)
+                messages = self.cached_messages
+                self.cached_messages = []
+            self.load_messages_to_cache(batch_size * 2 - pulled_messages_size)
+            return messages
```

Comment:
This is called only if prefetch is true and the cached_messages length is bigger then 0, so this is never called, I don't see where this is initialized for the first time

Reply:
You are correct, fixed the bug now